samples/DotNet/Microsoft.Azure.ServiceBus/BasicSessionSendReceiveUsingQueueClient/Program.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace BasicSessionSendReceiveUsingQueueClient
+{
+    using Microsoft.Azure.ServiceBus;
+    using Microsoft.Azure.ServiceBus.Primitives;
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    class Program
+    {
+        // Connection String for the namespace can be obtained from the Azure portal under the 
+        // 'Shared Access policies' section.
+        const string ServiceBusConnectionString = "{ServiceBus connection string}";
+        const string QueueName = "{Queue Name of a Queue that supports sessions}";
+        static IQueueClient queueClient;
+
+        static void Main(string[] args)
+        {
+            MainAsync().GetAwaiter().GetResult();
+        }
+
+        static async Task MainAsync()
+        {
+            const int numberOfSessions = 5;
+            const int numberOfMessagesPerSession = 3;
+
+            queueClient = new QueueClient(ServiceBusConnectionString, QueueName);
+
+            Console.WriteLine("======================================================");
+            Console.WriteLine("Press any key to exit after receiving all the messages.");
+            Console.WriteLine("======================================================");
+
+            // Register Session Handler and Receive Session Messages
+            RegisterOnSessionHandlerAndReceiveSessionMessages();
+
+            // Send messages with sessionId set
+            await SendSessionMessagesAsync(numberOfSessions, numberOfMessagesPerSession);
+
+            Console.ReadKey();
+
+            await queueClient.CloseAsync();
+        }
+
+        static void RegisterOnSessionHandlerAndReceiveSessionMessages()
+        {
+            // Configure the SessionHandler Options in terms of exception handling, number of concurrent sessions to deliver etc.
+            var sessionHandlerOptions =
+                    new SessionHandlerOptions(ExceptionReceivedHandler)
+                    {
+                        // Maximum number of Concurrent calls to the callback `ProcessSessionMessagesAsync`
+                        // Value 2 below indicates the callback can be called with 2 messages in parallel.
+                        // Set it according to how many messages the application wants to process in parallel.
+                        MaxConcurrentSessions = 2,
```

Comment:
Its worth to say that 2 is the number of sessions that are going to be active. They will receive only one message per session.